validate_response() fails when 'pk' path parameter coerced by DRF

By default, DRF [seems to change `pk` path parameters to the actual primary key field name](https://www.django-rest-framework.org/api-guide/settings/#schema_coerce_path_pk) for the schema. This leads to a strange error when the `SchemaTester`:

```py
>>> response = client.get("/api/names/123/")
>>> schema_tester.validate_response(response)
openapi_tester.exceptions.UndocumentedSchemaSectionError: Error: Unsuccessfully tried to index the OpenAPI schema by `/api/names/{pk}/`. 

For debugging purposes, other valid routes include: 

    • /api/names/
    • /api/names/{id}/
    • ...
```

When I set `REST_FRAMEWORK["SCHEMA_COERCE_PATH_PK"] = False` in my settings, it works, but then the schema uses the path parameter `{pk}` instead of `{id}`, which is not very nice.

My `Name` model's primary key field name is `id`, and my ViewSet generating the routes looks like this:

```py
class NameViewSet(viewsets.ReadOnlyModelViewSet):
    queryset = models.Name.objects.all()
    serializer_class = serializers.NameSerializer

router = SimpleRouter()
router.register(r"names", NameViewSet)
```

(Also possible that I've configured something incorrectly, since a lot has changed in the package recently...)

That's awesome, thanks a lot!

I updated to 1.3.1, and now I get a `ValueError` at this line: https://github.com/snok/drf-openapi-tester/blob/81a8479383ced390733a94ec68769559fa7ac751/openapi_tester/loaders.py#L151

```
venv/lib/python3.8/site-packages/openapi_tester/schema_tester.py:368: in validate_response
    response_schema = self.get_response_schema_section(response)
venv/lib/python3.8/site-packages/openapi_tester/schema_tester.py:118: in get_response_schema_section
    parameterized_path, _ = self.loader.resolve_path(response.request["PATH_INFO"], method=response_method)
venv/lib/python3.8/site-packages/openapi_tester/loaders.py:210: in resolve_path
    de_parameterized_path, resolved_path = super().resolve_path(endpoint_path=endpoint_path, method=method)
venv/lib/python3.8/site-packages/openapi_tester/loaders.py:136: in resolve_path
    path, resolved_route = self.handle_pk_parameter(
venv/lib/python3.8/site-packages/openapi_tester/loaders.py:153: in handle_pk_parameter
    imported_module = import_module(".".join(split_view_path))
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1011: in _gcd_import
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = '', package = None, level = 0

>   ???
E   ValueError: Empty module name

<frozen importlib._bootstrap>:950: ValueError
```

The `resolved_route` looks like this: `ResolverMatch(func=pench.api.views.NameViewSet, args=(), kwargs={'pk': '{id}'}, url_name=name-detail, app_names=[], namespaces=[], route=api/names/(?P<pk>[^/.]+)/$)` (`view_name` is also `'name-detail'`)

If it's of any help, I could try reproducing the error in the `test_project` and writing a test for it? Also haven't tried the i18n parameter yet, but I will as soon as possible.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 # Changelog
 
+## v1.3.2 2020-02-20
+
+* Hotfix regression due to pk resolution logic
 
 ## v1.3.1 2020-02-20
 

--- a/openapi_tester/loaders.py
+++ b/openapi_tester/loaders.py
@@ -3,7 +3,6 @@ import difflib
 import json
 import pathlib
 from functools import cached_property
-from importlib import import_module
 from json import dumps, loads
 from typing import Callable, Dict, List, Optional, Tuple
 from urllib.parse import ParseResult, urlparse
@@ -146,11 +145,7 @@ class BaseSchemaLoader:
     def handle_pk_parameter(  # pylint: disable=no-self-use
         self, resolved_route: ResolverMatch, path: str, method: str
     ) -> Tuple[str, ResolverMatch]:
-        split_view_path = resolved_route.view_name.split(".")
-        view_name = split_view_path.pop()
-        imported_module = import_module(".".join(split_view_path))
-        view = getattr(imported_module, view_name)
-        coerced_path = BaseSchemaGenerator().coerce_path(path=path, method=method, view=view)
+        coerced_path = BaseSchemaGenerator().coerce_path(path=path, method=method, view=resolved_route.func)  # type: ignore
         pk_field_name = "".join(
             list([entry.replace("+ ", "") for entry in difflib.Differ().compare(path, coerced_path) if "+ " in entry])
         )

--- a/openapi_tester/type_declarations.py
+++ b/openapi_tester/type_declarations.py
@@ -1,6 +1,6 @@
 """ Type declarations Module - this file is used for type checking and type casting """
 # flake8: noqa
-# pylint: disable=unused-import, invalid-name, unused-argument, no-self-use, missing-function-docstring
+# pylint: disable=unused-import, missing-function-docstring
 
 import sys
 from typing import TYPE_CHECKING

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "drf-openapi-tester"
-version = "1.3.1"
+version = "1.3.2"
 description = "Django test utility for validating OpenAPI response documentation"
 authors = ["Sondre Lillebø Gundersen <sondrelg@live.no>", "Na'aman Hirschfeld <nhirschfeld@gmail.com>"]
 maintainers = ["Na'aman Hirschfeld <nhirschfeld@gmail.com>"]
@@ -86,13 +86,17 @@ disable = """
     missing-function-docstring,
     import-outside-toplevel,
     fixme,
-    too-many-locals
+    line-too-long,
 """
+enable = "useless-suppression"
 
-[tool.pylint.BASIC]
+[tool.pylint.DESIGN]
 max-args = 6
 max-returns = 21
 max-branches = 20
+max-locals = 20
+
+[tool.pylint.BASIC]
 good-names = "_,e"
 
 [tool.coverage.run]

--- a/tests/test_schema_tester.py
+++ b/tests/test_schema_tester.py
@@ -32,6 +32,7 @@ from tests import example_object, example_schema_types
 from tests.utils import TEST_ROOT, iterate_schema, mock_schema, response_factory
 
 tester = SchemaTester()
+name_id = 1234567890
 parameterized_path = "/api/{version}/cars/correct"
 de_parameterized_path = "/api/v1/cars/correct"
 method = "get"
@@ -102,17 +103,13 @@ def test_loader_inference(settings):
         SchemaTester()
 
 
-def test_drf_coerced_model_primary_key(db, client):
-    name = Names.objects.create(custom_id_field=1234567890)
-    response = client.get(f"/api/v1/{name.custom_id_field}/names")
+@pytest.mark.parametrize(
+    "url", [f"/api/v1/{name_id}/names", "/api/v1/router_generated/names/", f"/api/v1/router_generated/names/{name_id}/"]
+)
+def test_drf_coerced_model_primary_key(db, client, url):
+    Names.objects.create(custom_id_field=name_id)
     schema_tester = SchemaTester()
-    schema_tester.validate_response(response)
-
-    response = client.get("/api/v1/router_generated/names/")
-    schema_tester = SchemaTester()
-    schema_tester.validate_response(response)
-    response = client.get(f"/api/v1/router_generated/names/{name.custom_id_field}/")
-    schema_tester = SchemaTester()
+    response = client.get(url)
     schema_tester.validate_response(response)
 
 

